nixos/modules/services/computing/boinc/client.nix

```diff
@@ -6,6 +6,24 @@ let
   cfg = config.services.boinc;
   allowRemoteGuiRpcFlag = optionalString cfg.allowRemoteGuiRpc "--allow_remote_gui_rpc";
 
+  # A package providing the boinc_client that will actually be executed.
+  execPkg = if cfg.useFHSEnv then
+    pkgs.buildFHSUserEnv {
+      name = "boinc_client";
+      targetPkgs = p: [ cfg.package ] ++ optional cfg.virtualbox.enable cfg.virtualbox.package;
+      multiPkgs = p: optional cfg.gpu.nvidia.enable cfg.gpu.nvidia.package;
+      runScript = "/bin/boinc";
+      extraBuildCommands = ''
+        if [ -e lib/vendors ]; then
+          chmod +w etc
+          mkdir -p etc/OpenCL
+          ln -s $(realpath lib/vendors) etc/OpenCL/vendors
+          chmod -w etc
+        fi
+      '';
```

Comment:
@abbradar awesome, thanks. It'll still need to optionally depend on `nvidia_x11` since the CUDA libraries are required as well for NVIDIA support, but I'll make OpenCL support a separate option that pulls in `ocl-icd`. This new option ought to cover AMD and Intel GPUs as well.